### DISCUSSION TITLE
Containerization and Separating the Frontend and Backend.

### DISCUSSION THREAD

**Discussion by dopry** [bpmnServer/bpmn-server#126] [Ideas]
Following the getting started is a little awkward with the whole `npm install bpmn-server` and `cp -r node_modules/...`.

If the bpmn-server were containerized there would be a dedicated WebApp (I think this may be what is happening in the bpnm-client folder). The WebApp repo could be setup with a docker-compose file to launch mongo and npm-server... making the process to setup a local demo more like...

```
git clone git://.../front-end
docker compose up -d
npm install
npm start
```

**Comment by ralphhanna** [comment]
Thanks for the feedback.

I am working on separating the front-end as a separate repo

I think what you are suggesting is something like this:
1) Repo bpmn-server
2) Repo bpmn-web - requires bpmn-server
3) Repo bpmn-client

### installation of bpmn-server along with bpmn-web
```js
git clone .../bpmn-web 
npm install
copy sample.env .env
```
edit .env file
```js
npm run setup
npm start
```

#### To Update
```js
git pull
npm update

```

### installation of bpmn-client
create a folder or use an existing project
```js
git clone .../bpmn-client-sample 
copy sample.env .env
```
edit .env file
```js
npm install
```
#### To Update
```js
git pull
npm update

```

**Comment by dopry** [comment]
more or less. If there were a separate repo for webapp, I can make pretty quick work of that kind of change.

**Comment by dopry** [comment]
I woulnd't copy bpmn-web into the bpnm-server repo. If you want to do live development against another project there is npm link. If you wanted to work with the web app or quick start it, it should just npm install bpmn-server.

**Comment by ralphhanna** [comment]
I agree

Sent from my iPhone

On Nov 1, 2023, at 4:59 PM, dopry ***@***.***> wrote:

﻿

I woulnd't copy bpmn-web into the bpnm-server repo. If you want to do live development against another project there is npm link. If you wanted to work with the web app or quick start it, it should just npm install bpmn-server.

**Reply by kstan79** [reply]
containerize bpmn-server and put into docker hub is good approach, it can make this project much more popular and give dumps up 100%. 

I think after nestjs supported, documentation tidy up may make it real.

**Comment by ralphhanna** [comment]
docker is now available [https://github.com/bpmnServer/docker](https://github.com/bpmnServer/docker)
